backend/src/controllers/v2/secretsController.ts

```diff
@@ -0,0 +1,472 @@
+            _id: new Types.ObjectId(),
+            secret: _id,
+            version,
+            workspace,
+            type,
+            user,
+            environment,
+            isDeleted: false,
+            secretKeyCiphertext,
+            secretKeyIV,
+            secretKeyTag,
+            secretKeyHash,
+            secretValueCiphertext,
+            secretValueIV,
+            secretValueTag,
+            secretValueHash
+        }))
+    });
+
+    const addAction = await EELogService.createActionSecret({
+        name: ACTION_ADD_SECRETS,
+        userId: req.user._id.toString(),
+        workspaceId,
+        secretIds: newSecrets.map((n) => n._id)
+    });
+
+    // (EE) create (audit) log
+    addAction && await EELogService.createLog({
+        userId: req.user._id.toString(),
+        workspaceId,
+        actions: [addAction],
+        channel,
+        ipAddress: req.ip
+    });
+
+    // (EE) take a secret snapshot
+    await EESecretService.takeSecretSnapshot({
+        workspaceId
+    });
+
+    if (postHogClient) {
+        postHogClient.capture({
+            event: 'secrets deleted',
+            distinctId: req.user.email,
+            properties: {
+                numberOfSecrets: toAdd.length,
+                environment,
+                workspaceId,
+                channel: req.headers?.['user-agent']?.toLowerCase().includes('mozilla') ? 'web' : 'cli',
+                userAgent: req.headers?.['user-agent']
+            }
+        });
+    }
+
+    return res.status(200).send({
+        secrets: newSecrets
+    });
+}
+
+/**
+ * Return secret(s) for workspace with id [workspaceId], environment [environment] and user
+ * with id [req.user._id]
+ * @param req 
+ * @param res 
+ * @returns 
+ */
+export const getSecrets = async (req: Request, res: Response) => {
+    const { workspaceId, environment } = req.query;
+    
+    let userId: Types.ObjectId | undefined = undefined // used for getting personal secrets for user
+    if (req.user) {
+        userId = req.user._id;
+    }
+
+    if (req.serviceTokenData) {
+        userId = req.serviceTokenData.user._id
+    }
+    
+    const [err, secrets] = await to(Secret.find(
+        {
+            workspace: workspaceId,
+            environment,
+            $or: [
+                { user: userId }, 
+                { user: { $exists: false } }
+            ],
+            type: { $in: [SECRET_SHARED, SECRET_PERSONAL] }
+        }
+    ).then())
+
+    if (err) throw ValidationError({ message: 'Failed to get secrets', stack: err.stack });
+    
+    const channel = req.headers?.['user-agent']?.toLowerCase().includes('mozilla') ? 'web' : 'cli';
+    
+    const readAction = await EELogService.createActionSecret({
+        name: ACTION_READ_SECRETS,
+        userId: req.user._id.toString(),
+        workspaceId: workspaceId as string,
+        secretIds: secrets.map((n: any) => n._id)
+    });
+
+    readAction && await EELogService.createLog({
```

Comment:
Can this code be simplified and take advantage of Promises?
```ts
await EELogService.createActionSecret({
        name: ACTION_READ_SECRETS,
        userId: req.user._id.toString(),
        workspaceId: workspaceId as string,
        secretIds: secrets.map((n: any) => n._id)
    })
    .then((readAction)=>{
        EELogService.createLog({
             userId: req.user._id.toString(),
             workspaceId: workspaceId as string,
             actions: [readAction],
             channel,
             ipAddress: req.ip
        });
    })
    .then(()=>{return res.status(200).json({secrets})})
```

or just be entirely async without await, it will speed up response time because response is not dependent to Audit Logs success status.

Reply:
Ah yea we can do without await then because you're right response is not dependent on the audit log success status — Let's submit that in the next PR